### PR TITLE
1.23.0 Release -> Master

### PR DESCRIPTION
**Pull request by mchowning**
Release for Gutenberg Mobile v1.23.0

## Related PRs

WPAndroid: https://github.com/wordpress-mobile/WordPress-Android/pull/11331
WPiOS: https://github.com/wordpress-mobile/WordPress-iOS/pull/13485

FluxC-Android: N/A
Aztec-Android: N/A
Aztec-iOS: N/A

Gutenberg: TBD
WPiOS after: TBD
WPAndroid after: TBD

Other repos: No changes in other repos
Merge to gutenberg-mobile `master` to `develop`: TBD

## Extra PRs that Landed After the Release Was Cut

#1894 Fix inserter
#1930 Make sure strong tags are removed from title
#1938 Fix alignment issues on iOS

## Changes
Gathered via `git log v1.22.0..release/1.23.0 --first-parent --oneline` and trimmed based on whether the change is functional.

* #1914 Fix menu item width inside the inserter
*  #1857 Update aztec to 1.16.0
* #1918 Correct add icon appearance
* #1904 Removing DEV flag for Image block size options
* #1879 Fix link button on Rich Text
* #1895 Image upload preview incorrect on id conflict
* #1616 Button Block
* #1866 Enable Button block in production
* #1855 Enable gallery upload options in production
* #1837 Correct layout within bottom-sheet
* #1834 Replace Slider with Stepper for columns settings on Gallery block
* #1859 Feat: refine block borders
*  #1872 Fix layout issue in Shortcode
* #1850 Feat: Open Group block for production
* #1838 Adjust vertical margins in InnerBlock
*  #1798 Fix gallery size labels

## PR Submission Checklist

- [X] I have considered adding unit tests where possible.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: RELEASE-NOTES.txt [hunk 1]
```diff
@@ -1,5 +1,16 @@
...
+* Add support for upload options in Gallery block
+* Add support for size options in the Image block
+* New block: Button
+* Add scroll support inside block picker and block settings
+* [Android] Fix issue preventing correct placeholder image from displaying during image upload
+* [iOS] Fix diplay of large numbers on ordered lists
+
 1.22.0
 ------
+* Add support for changing images size in Gallery block
```
Comment (mchowning) [inline RELEASE-NOTES.txt]: :wave: @mkevins ! Looks like this should be moved up to under `1.23.0`. Does that seem right to you?

Comment (mkevins) [reply RELEASE-NOTES.txt]: Actually, that was released as part of 1.22.0, so it's correct there, but I just noticed it's listed twice under 1.22.0 :thinking: I guess one of the duplicates under 1.22.0 can be removed.

**Comment by geriux** [comment]
**Sanity tests (Android)**

MediaText-1

- [ ] Media Text block - Insert image from device (failing) - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/media-text.md#tc001-i)
- [ ] Media Text block - Close/Re-open post with an ongoing image upload - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/media-text.md#tc003-i)

MediaText-2

- [ ] Media Text block - Insert video from device (cancel) - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/media-text.md#tc002-v)
- [ ] Media Text block - Insert image from device (cancel) - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/media-text.md#tc002-i)

MediaText-3

- [ ] Media Text block - Insert video from device (failing) - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/media-text.md#tc001-v)
- [ ] Media Text block - Close/Re-open post with an ongoing video upload - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/media-text.md#tc003-v)

MediaText-4

- [ ] Media Text block - Media & Text alignment - steps
- [ ] Media Text block - Vertical alignment - steps

MultipleUpload-1

- [ ] Simultaneous uploads - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/media-interaction.md#tc001)

Shortcode-1

- [ ] Shortcode block - Add a youtube link - steps

**Comment by chipsnyder** [comment]
Gallery-1
- ❌Gallery block - Close/Re-open post with an ongoing image upload - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/gallery.md#tc001)
    - Issue: https://github.com/wordpress-mobile/gutenberg-mobile/issues/1928

- [x] Gallery block - Close post with an ongoing image upload - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/gallery.md#tc002)

Gallery-2

- [ ] Gallery block - Insert image from device (failing) - steps
- [ ] Gallery block - Insert image from device (cancel) - steps

Gallery-3

- [ ] Gallery block - Add Gallery Caption - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/gallery.md#tc003)
- [ ] Gallery block - Add Gallery Image Caption - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/gallery.md#tc004)
- [ ] Gallery block - Try adding same images from WP Media library and moving the images around - steps

Gallery-4

- [ ] Gallery block - Settings: Link to - steps
- [ ] Gallery block - Settings: Column number - steps
- [ ] Gallery block - Settings: Crop images - steps

**Comment by lukewalczak** [comment]
**Sanity tests (iOS)**

Button-1

- [ ] Button block - Render custom text color - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/button.md#tc011)
- [ ] Button block - Render custom background color - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/button.md#tc010)
- [ ] Button block - Render gradient background color - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/button.md#tc012)
- [ ] Button block - Check if selection / caret color matches font color - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/button.md#tc015)

Button-2

- [ ] Button block - Edit text styles - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/button.md#tc003)
- [ ] Button block - Fallback to default colors in case theme colors are used - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/button.md#tc013)
- [ ] Button block - New Button is created with the gray color - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/button.md#tc014)
- [ ] Button block - Settings: Change Button border radius - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/button.md#tc009)
- [ ] Button block - The newly created button has set background color to the theme-default color in preview - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/button.md#tc016)
- [ ] Button block - Tapping `delete` key removes block when Button is empty - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/button.md#tc017)

Button-3

- [ ] Button block - Settings: Open in new tab - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/button.md#tc005)
- [ ] Button block - Settings: Link rel - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/button.md#tc006)
- [ ] Button block - Settings: Link URL - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/button.md#tc004)
- [ ] Button block - Settings: Remove link - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/button.md#tc008)
- [ ] Button block - Settings: Synchronize with button options - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/button.md#tc007)

Button-4

- [ ] Button block - Link in the clipboard is automaticially added into the empty URL field - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/button.md#tc002)
- [ ] Button block - Toolbar link button is active when Button has link - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/button.md#tc018)
- [ ] Button block - Button max width is calculated OK inside inner blocks(iOS only) - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/button.md#tc001)

**Comment by jbinda** [comment]
**Sanity test (iOS)**

Group-1

- [x] Group - AppenderButton is rendered - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/group.md#tc001)
- [ ] Group - Deep nesting is possible (iOS only) - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/group.md#tc002)
  ❌ **After add 6th level of nesting the placeholder stops to be visible**
 
- [x] Group - Check if Group placeholder is visible for the unselected state - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/group.md#tc003)
- [x] Group - Check if Group placeholder is render in nested structure - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/group.md#tc004)
- [x] Group - Nested block have proper border styling - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/group.md#tc005)
- [x] Group - Nested block have proper margins values - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/group.md#tc006)
- [x] Group - Nested selection cause applying dimmed style on the rest of blocks - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/group.md#tc007)
- [x] Group - Breadcrumbs on FloatingToolbar is properly displayed - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/group.md#tc008)
- [x] Group - Navigation up button works as expected - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/group.md#tc009)
- [x] Group - Navigation down works according to parent-first approach - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/group.md#tc010)
- [x] Group - Cross navigation between blocks works as expected - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/group.md#tc011)
- [x] Group - Ungroup button works as expected - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/group.md#tc012)
- [x] Group - Check if in DarkMode all components gets proper colors (iOS) - [steps](https://github.com/wordpress-mobile/test-cases/blob/master/test-cases/gutenberg/group.md#tc013)

DarkMode-1(iOS)

- [ ] Gallery block - Dark mode - steps
- [ ] Shortcode block - Dark mode - steps
- [ ] Media Text block - Dark mode - steps

DarkMode-2(iOS)

- [ ] Spacer block - Dark mode - steps
- [ ] Button block - Dark mode - steps
- [ ] Group - Dark mode - steps

Spacer-1

- [ ] Spacer block - Settings: Height in pixels - steps


@pinarol 
Apart of above test results I have notice one more thing. When the `Group` has only 1 item inside  I would expect the selection after remove that block goes to `Group` as a parent. It doesn't - currently there is no block selected after proceed with above scenario.

When Group has more than one item the selection goes to block before removed one which I think is something that we expect.

**Comment by pinarol** [comment]
> Apart of above test results I have notice one more thing. When the Group has only 1 item inside I would expect the selection after remove that block goes to Group as a parent. It doesn't - currently there is no block selected after proceed with above scenario.

Thanks for reporting that. It is not a regression or a bug. Actually that looks pretty expected to me, I tried the same scenario and it felt natural. We can do some fine tuning about that kind of details at some point but it is a pretty new block, let’s give it some time and see what kind of user feedback we’ll have soon.

**Comment by pinarol** [comment]
> After add 6th level of nesting the placeholder stops to be visible

@jbinda I'd consider that an edge case but still, let's track it with an issue, could you be able to open one?

**Comment by jbinda** [comment]
> > After add 6th level of nesting the placeholder stops to be visible
> 
> @jbinda I'd consider that an edge case but still, let's track it with an issue, could you be able to open one?

Ok, I have open issue here (#1935) and update test checklist as well

**Comment by mchowning** [comment]
I tested all the changes with the latest code and everything looks good to me.

The issue with closing and reopening a post while an image is uploading (#1928) was determined to be enough of an edge case that it did not need to be fixed for this release.

Likewise, I agree with @pinarol 's [opinion](https://github.com/wordpress-mobile/gutenberg-mobile/pull/1924#issuecomment-588262843) that the placeholder issue with deeply nested blocks (#1935) is an edge case that does not need to be addressed for this release.
